### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by RoboticMind**
* Update some of the dependency versions to work with Flutter `v3.10.5`
* ~~Change paths in the pubspec.yaml from `/opt/devel/uva/repos/*` to `/workspaces/*` to match the container~~

### CODE REVIEW

File: sequentialcompose/impl/frontend/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux [hunk 1]
```diff
@@ -0,0 +1 @@
+/root/.pub-cache/hosted/pub.dev/url_launcher_linux-3.0.5/
```
Comment (nicholasphair) [inline sequentialcompose/impl/frontend/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux:1]: I don't think we want to support linux desktop apps. So any `frontend/linux` directories should be removed.

**Review by nicholasphair — CHANGES_REQUESTED** [review]
Thanks for this. Once there is an updated docker image to test on I can try it out and merge.

File: sequentialcompose/impl/frontend/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux (hunk 1 above)
Comment (RoboticMind) [reply sequentialcompose/impl/frontend/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux]: Whoops, forgot that I had accidentally added the Linux Desktop platform at one point. I think I got all the relevant files/folders deleted now

**Comment by RoboticMind** [comment]
Now made a PR for updating the docker build https://github.com/kevinsullivan/moralpain_config/pull/15

Also a bit clunky, but you can automate the process of fixing up the packages folder by running this command that finds all package folders and cleans and reinstalls the depends 

```bash
find . -name "packages" -execdir sh -c "cd packages && flutter clean && flutter pub get" \;
```

**Comment by RoboticMind** [comment]
Hmm something strange I just noticed. Each app separately runs without issue but running the sequential compose generates this runtime error once things are started:

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following ProviderNotFoundException was thrown building ThermometerLabel<ThermometerCubit>:
Error: Could not find the correct Provider<ThermometerCubit> above this
BlocBuilder<ThermometerCubit, double> Widget
```

Here's how it shows on the page itself:
image

**Comment by nicholasphair** [comment]
As I think more about this, I think the right path for now is to use relative paths. Could we separate this into two prs. One for the version upgrades and the other for that paths?

**Comment by RoboticMind** [comment]
Removed the path changes from this PR now, will create another for relative PRs in a bit

**Pull request merged**